drivers/sensor/lsm6dsv16x/lsm6dsv16x.c

```diff
@@ -83,8 +83,8 @@ static int lsm6dsv16x_accel_range_to_fs_val(int32_t range)
 	return -EINVAL;
 }
 
-static const uint16_t lsm6dsv16x_gyro_fs_map[] = {250, 125, 500, 0, 1000, 0, 2000};
-static const uint16_t lsm6dsv16x_gyro_fs_sens[] = {2, 1, 4, 0, 8, 0, 16};
+static const uint16_t lsm6dsv16x_gyro_fs_map[] = {125, 250, 500, 1000, 2000, 4000};
+static const uint16_t lsm6dsv16x_gyro_fs_sens[] = {1, 2, 4, 8, 16, 32};
```

Comment:
Good catch, but 4000 dps is wrong (it's 0xc and not 0x5).
So, I would say:

```
static const uint16_t lsm6dsv16x_gyro_fs_map[] = {125, 250, 500, 1000, 2000, 0, 0, 0, 0, 0, 0, 0, 4000};
static const uint16_t lsm6dsv16x_gyro_fs_sens[] = {1, 2, 4, 8, 16, ,0, 0, 0, 0, 0, 0, 0, 32};
```

The index 'i' should result in 12 (0xc) for 400 dps.
Does it make sense?

Reply:
Thanks for the review @avisconti and thanks for pointing out the `0xc` binding mismatch!

I see that the binding enums used to be sequential prior to commit e72baabe910.  Is there a reason the enum for the 4000DPS range was changed from `5` to `0xc`?  i.e., can we change the binding enum back to `5` and leave the arrays as they are in this PR?

Reply:
> Thanks for the review @avisconti and thanks for pointing out the `0xc` binding mismatch!
> 
> I see that the binding enums used to be sequential prior to commit [e72baab](https://github.com/zephyrproject-rtos/zephyr/commit/e72baabe910934564b4bac2c4a8ec59867b84701). Is there a reason the enum for the 4000DPS range was changed from `5` to `0xc`? i.e., can we change the binding enum and `LSM6DSV16X_DT_FS_4000DPS` macro back to `5`, and leave the arrays contiguous as they are in this PR?

We cannot change it as that's the way the h/w is done. Please look at https://www.st.com/resource/en/datasheet/lsm6dsv16x.pdf paragraph 9.19

Reply:
Ah, I see.  The array's **index** is being used to determine the value of the register, rather than the value.

Wouldn't it make more sense to put the actual register **value** in the array? After all, imagine you had to set a register value to something like 0x80 - it would be unwieldy to create an array with more than a hundred 0's just to get to the last index.

So the range array could look like this:
```
static const uint16_t lsm6dsv16x_gyro_fs_map[] = {0, 1, 2, 3, 4, 0x0c};
```

And the binding enum would just be used as the index into the array to get the value.  Does that make more sense?

Reply:
> Ah, I see. The array's **index** is being used to determine the value of the register, rather than the value.
> 
> Wouldn't it make more sense to put the actual register **value** in the array? After all, imagine you had to set a register value to something like 0x80 - it would be unwieldy to create an array with more than a hundred 0's just to get to the last index.
> 
> So the range array could look like this:
> 
> ```
> static const uint16_t lsm6dsv16x_gyro_fs_map[] = {0, 1, 2, 3, 4, 0x0c};
> ```
> 
> And the binding enum would just be used as the index into the array to get the value. Does that make more sense?

Mmmh, no, you can't do that because the value received from SENSOR_ATTR_FULL_SCALE is a value in mdps

Reply:
> Mmmh, no, you can't do that because the value received from SENSOR_ATTR_FULL_SCALE is a value in mdps

I'm sorry, I don't see how that has any bearing on how the enums are mapped to the values used for the register setting?

Reply:
Ah OK yes I understand now - I just pushed your suggested fix in 26d1249c711a29ef4a5eff15f0886be94d32c8be